test/AST-Quake/cphase.cpp

```diff
@@ -0,0 +1,46 @@
+/*************************************************************** -*- C++ -*- ***
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ *******************************************************************************/
+
+// RUN: cudaq-quake %s | FileCheck %s
+
+#include <cudaq.h>
+
+// Verifies that cphase op produces the same IR as r1<cudaq::ctrl>
+struct CphaseOp {
+  void operator()() __qpu__ {
+    cudaq::qubit q1, q2;
+    cphase(1.234, q1, q2);
+  }
+};
+
+// clang-format off
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__CphaseOp() attributes
+// CHECK:           %[[VAL_0:.*]] = quake.alloca : !quake.qref
+// CHECK:           %[[VAL_1:.*]] = quake.alloca : !quake.qref
+// CHECK:           %[[VAL_2:.*]] = arith.constant {{.*}} : f64
+// CHECK:           quake.r1 [%[[VAL_0]] : !quake.qref] |%[[VAL_2]] : f64|(%[[VAL_1]])
+// CHECK:           return
+// CHECK:         }
+// clang-format on
+
+struct CtrlR1 {
+  void operator()() __qpu__ {
+    cudaq::qubit q1, q2;
+    r1<cudaq::ctrl>(1.234, q1, q2);
```

Comment:
Might as well add a `p` test here, eh?

Reply:
I wanted to as well, but `p` is not 'declared' in `qubit_qis.h`, hence will trigger a compile error.
If you want, I can add `p` to `qubit_qis.h`. What do you think?

Reply:
There is an open question we've had here as to whether we want `p` or `cphase`. Early on we decided against `p` in favor of `r1`, and since we generate controlled operations via the single target operations, we might want to avoid `cphase` in favor of always using `r1<cudaq::ctrl>`. I want @boschmitt to chime in here. 

So it might be that instead of adding `cphase` to the bridge, we should have removed it from `qubit_qis.h`.

Reply:
This is how I was thinking too. We should remove the cphase function from qubit_qis.h. @1tnguyen what do you think? Would you want to back out those bridge changes and instead remove cphase from qubit_qis.h? How does this affect your work?

Reply:
+1

Reply:
Thanks for the explanation @boschmitt! 


> what do you think? How does this affect your work?

@amccaskey Totally agreed with having a single `r1` op to make API cleaner/less overloaded.
Actually, this doesn't affect my work at all. I was writing some CUDAQ kernels and looking for this gate. A quick search indicated that it was valid (used in some other places/tests in the codebase), but not handled in the bridge. I've since changed my own code to use `r1<ctrl>` but figured that it might be worth submitting a PR :-)

> Would you want to back out those bridge changes and instead remove `cphase` from `qubit_qis.h`?

Definitely. I'll update this PR accordingly.